client-core/src/service/wallet_service.rs

```diff
@@ -329,35 +520,17 @@ pub fn load_wallet<S: SecureStorage>(
         // storage -> wallet
         let info_keyspace = get_info_keyspace(name);
         new_wallet.view_key = read_pubkey_enc(storage, &info_keyspace, "viewkey", enckey)?;
-        // pubkey
-        let info_keyspace = format!("{}_{}_info", KEYSPACE, name);
-        let staking_keyspace = get_stakingkey_keyspace(name);
-        let stakingkey_count: u64 =
-            read_number(storage, &info_keyspace, "stakingkeyindex", Some(0))?;
-        for i in 0..stakingkey_count {
-            let stakingkey = read_pubkey(storage, &staking_keyspace, &format!("{}", i))?;
-            new_wallet.staking_keys.insert(stakingkey);
-        }
-
-        // roothash
-        let roothash_keyspace = get_roothash_keyspace(name);
-        let roothash_count: u64 = read_number(storage, &info_keyspace, "roothashindex", Some(0))?;
-        for i in 0..roothash_count {
-            let value = storage.get(&roothash_keyspace, format!("{}", i))?;
-            if let Some(raw_value) = value {
-                let mut roothash_found: H256 = H256::default();
-                roothash_found.copy_from_slice(&raw_value);
-                new_wallet.root_hashes.insert(roothash_found);
-            }
-        }
-
         // load walletkind
         let walletkind: u64 = read_number(storage, &info_keyspace, "walletkind", Some(0))?;
         new_wallet.wallet_kind = walletkind.into();
+        // set name, enckey
+        new_wallet.name = name.into();
+        new_wallet.enckey = Some(enckey.clone());
+        let newstorage = storage.clone();
+        new_wallet.wallet_storage = Some(Arc::new(Mutex::new(WalletStorageImpl::new(newstorage))));
```

Comment:
the code structure is a bit deviated from original intension, i guess it takes a bigger refactor to make it better ;D